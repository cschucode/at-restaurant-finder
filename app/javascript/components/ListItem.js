import React, { useState } from 'react';

import ReactStars from 'react-rating-stars-component';

import './ListItem.scss';

const ListItem = ({ ...props }) => {
  const [isFavorite, setIsFavorite] = useState(false);
  const { place, tooltip, onMouseEnter, onMouseLeave, isSelected } = props;

  return (
    <div onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave} className={'list-item' + (tooltip ? ' tool-tip' : '') + (isSelected ? ' selected' : '')} >
      <div className="list-item__image" style={{backgroundImage: `url(${place.icon})`}}></div>
      <div className="list-item__details">
        <p className="list-item__name">{place.name}</p>
        <div className="list-item__rating">
          <ReactStars
            count={5}
            edit={false}
            value={Math.round(place.rating)}
            size={16}
            activeColor="#ffd700"
          />
          <span className="list-item__rating-count">({place.user_ratings_total})</span>
        </div>
        <div>
          {place.price_level ? <span className="list-item__pricing">{getPriceLevel(place.price_level)}<span> &#183; </span></span> : ''}
          <span className="list-item__info">{place.vicinity || place.formatted_address}</span>
        </div>
        <button className={"list-item__favorite-icon" + (isFavorite ? ' favorite' : '')} onClick={() => setIsFavorite(!isFavorite)} />
      </div>
    </div>
  );
}

const getPriceLevel = (level) => {
  const dollars = ['$', '$$', '$$$', '$$$$', '$$$$$'];
  return dollars[level];
}

export default ListItem;