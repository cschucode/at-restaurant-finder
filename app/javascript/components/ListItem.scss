.list-item {
  background-color: #fff;
  border: 1px solid #bbb;
  border-radius: 5px;
  display: flex;
  flex-direction: row;
  margin-bottom: 20px;
  padding: 15px;
  position: relative;

  &:hover {
    border: 1px solid green;
    transition: border-color 500ms;
  }

  &.tool-tip {
    border: none;
    margin-bottom: 0;
    padding: 5px;
  }

  &.tool-tip &__favorite-icon {
    display: none;
  }

  &__image {
    background-repeat: no-repeat;
    background-size: cover;
    border: 1px solid #bbb;
    border-radius: 3px;
    height: 60px;
    margin-right: 15px;
    width: 60px;
  }

  &__details {
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: space-between;
  }

  &__name {
    font-size: 14px;
    font-weight: 600;
    margin: 0;
    width: 200px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &__rating {
    display: flex;
    flex-direction: row;
    align-items: center;

    &-count {
      margin-left: 5px;
    }
  }

  &__rating-count,
  &__pricing,
  &__info {
    font-size: 12px;
  }

  &__favorite-icon {
    background: url('https://cdn-assets.alltrails.com/assets/packs/media/search/icon-action-favorite-dark-a05b1a78.svg') no-repeat 50%;
    border: none;
    height: 24px;
    width: 24px;
    position: absolute;
    right: 15px;
    top: 15px;
    transition: background 500ms;

    &.favorite {
      background: url('https://cdn-assets.alltrails.com/assets/packs/media/search/icon-action-favorite-solid-3a3f7aa4.svg') no-repeat 50%;
      transition: background 500ms;
    }
  }
}