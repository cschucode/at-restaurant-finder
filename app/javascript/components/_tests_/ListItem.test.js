import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from '../ListItem';

describe('<ListItem />', () => {
  test('Render ListItem component', () => {
    const mockPlace={
      name: 'Mock Restaurant',
      photos: [{
        getUrl: () => {
          return 'www.image.png'
        }
      }],
      rating: 3.8,
      user_ratings_total: 135,
      price_level: 2,
      vicinity: '123 Lombard Street',
      place_id: 'abc123',
    }

    const component = renderer.create(
      <ListItem
        place={mockPlace}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
