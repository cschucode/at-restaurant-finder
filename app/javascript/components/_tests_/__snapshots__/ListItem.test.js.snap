// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> Render ListItem component 1`] = `
<div
  className="list-item"
>
  <div
    className="list-item__image"
    style={
      Object {
        "backgroundImage": "url(www.image.png)",
      }
    }
  />
  <div
    className="list-item__details"
  >
    <p
      className="list-item__name"
    >
      Mock Restaurant
    </p>
    <div
      className="list-item__rating"
    >
      <div
        className="react-stars-wrapper-"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          aria-label="add rating by typing an integer from 0 to 5 or pressing arrow keys"
          className=""
          onKeyDown={[Function]}
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
          tabIndex={null}
        >
          <p
            role="status"
            style={
              Object {
                "left": "-200rem",
                "position": "absolute",
              }
            }
          >
            0
          </p>
        </div>
      </div>
      <span
        className="list-item__rating-count"
      >
        (
        135
        )
      </span>
    </div>
    <div>
      <span
        className="list-item__pricing"
      >
        $$$
        <span>
           Â· 
        </span>
      </span>
      <span
        className="list-item__info"
      >
        123 Lombard Street
      </span>
    </div>
    <button
      className="list-item__favorite-icon"
      onClick={[Function]}
    />
  </div>
</div>
`;
